### PR TITLE
install-deps.sh: remove failing error catching

### PR DESCRIPTION
**Pull request by epuertat**
Fixes: https://tracker.ceph.com/issues/41013
Signed-off-by: Ernesto Puerta <epuertat@redhat.com>

- [x] References tracker ticket
- [x] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: install-deps.sh [hunk 1]
```diff
@@ -374,7 +374,6 @@ else
 	if [ -n "$dts_ver" ]; then
             ensure_decent_gcc_on_rh $dts_ver
 	fi
-        ! grep -q -i error: $DIR/yum-builddep.out || exit 1
```
Comment (tchaikov) [inline install-deps.sh]: can we move this line before the `if [ -n "$dts_ver" ]; then` block?

Comment (epuertat) [reply install-deps.sh]: In my environment `ensure_decent_gcc_on_rh` does barely anything (devtoolset 8 is pre-installed by Dockerfile, and it's not 8.3.1). I'll check it, but don't think it's going to change much :S

**Comment by epuertat** [comment]
@tchaikov I've brought the removed line back to here, but with previously filtering out the false positive (`"ValueError: ..."`). As it's hard to find out/reproduce what was the issue that motivated that `grep error:` trigger, and I don't see any major fixes in `yum-builddep` since 2014, I was not fully comfortable on removing this line. So, I prefer this ad-hoc fix rather than removing the whole thing.

**Comment by epuertat** [comment]
jenkins test make check

**Comment by epuertat** [comment]
jenkins test make check arm64

**Pull request merged**